types/context.go

```diff
@@ -156,6 +159,9 @@ func (c Context) WithBlockHeader(header cmtproto.Header) Context {
 	// https://github.com/gogo/protobuf/issues/519
 	header.Time = header.Time.UTC()
 	c.header = header
+
+	// when calling withBlockheader on a new context chainID in the struct is empty
+	c.chainID = header.ChainID
```

Comment:
Do we expect `WithChainID` to have the same behavior? (Set the chainID to the header?).
Currently in all SDK versions, it does not do it.

Reply:
tbh this is weird, the chainID is in the header and on the struct, but they werent updated together, I think this is two bugs, first storing it twice second not updating or checking the other when updating one of them